File: app/src/main/java/org/baiyu/fuckshare/filetype/OtherType.java
package org.baiyu.fuckshare.filetype;

public enum OtherType implements FileType {
    UNKNOWN(null);

    private final String extension;

    OtherType(String extension) {
        this.extension = extension;
    }

    @Override
    public String getExtension() {
        return extension;
    }
}

File: app/src/main/java/org/baiyu/fuckshare/filetype/OtherType.kt
package org.baiyu.fuckshare.filetype

enum class OtherType(
    override val extension: String?,
    override val signatures: Set<Map<Int, ByteArray>>?
) : FileType {
    UNKNOWN(null, null),
    PDF(
        "pdf", setOf(
            mapOf(
                0 to "%PDF-".toByteArray(Charsets.US_ASCII)
            )
        )
    ),
    M3U(
        "m3u", setOf(
            mapOf(
                0 to "#EXTM3U".toByteArray(Charsets.US_ASCII)
            )
        )
    );
}